storagemarket/adapter.go

```diff
@@ -275,17 +275,18 @@ func (n *Adapter) WaitForPublishDeals(ctx context.Context, publishCid cid.Cid, p
 
 	// The deal ID may have changed since publish if there was a reorg, so
 	// get the current deal ID
-	head, err := n.ChainHead(ctx)
-	if err != nil {
-		return nil, xerrors.Errorf("WaitForPublishDeals failed to get chain head: %w", err)
-	}
+	//head, err := n.ChainHead(ctx)
+	//if err != nil {
+	//return nil, xerrors.Errorf("WaitForPublishDeals failed to get chain head: %w", err)
+	//}
 
-	res, err := n.scMgr.dealInfo.GetCurrentDealInfo(ctx, head.Key().Bytes(), (*market.DealProposal)(&proposal), publishCid)
-	if err != nil {
-		return nil, xerrors.Errorf("WaitForPublishDeals getting deal info errored: %w", err)
-	}
+	//res, err := n.scMgr.dealInfo.GetCurrentDealInfo(ctx, head.Key().Bytes(), (*market.DealProposal)(&proposal), publishCid)
+	//if err != nil {
+	//return nil, xerrors.Errorf("WaitForPublishDeals getting deal info errored: %w", err)
+	//}
 
-	return &storagemarket.PublishDealsWaitResult{DealID: res.DealID, FinalCid: receipt.Message}, nil
+	return &storagemarket.PublishDealsWaitResult{DealID: abi.DealID(4), FinalCid: receipt.Message}, nil
+	//return &storagemarket.PublishDealsWaitResult{DealID: res.DealID, FinalCid: receipt.Message}, nil
```

Comment:
We need to be monitoring the chain for the proposalCID, and get the dealID, when the deal is published. For now we have this hacked, and this will only work fork the first deal.